### PR TITLE
Feature: Privacy Notice Consent popup for Firefox

### PR DESCRIPTION
**Pull request by Morpheus0x**
This pull request addresses the issue raised [here](https://github.com/orgs/ActivityWatch/discussions/818). This should satisfy the Mozilla addon review team. I have written a small Privacy Notice in `static/consent.html`, no idea if this is enough and if it can be used as a privacy policy for the whole addon...

I have added a saveguard to the normal addon popup page. If the user dismisses the privacy notice, the enable checkbox is disabled and a button is shown that shows the consent popup again.

I tested this with latest Firefox and Chromium.

### CODE REVIEW

**Comment by Morpheus0x** [comment]
Something I forgot to mention, the consent popup explicitly only shows up for Firefox and not for any other fork or Chromium based Browser. The extension will just work as it has previously, except for Firefox.

File: static/consent.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+"use strict";
+
+function consentListeners() {
+  let consent_refused = document.getElementById('consent-refused');
+  let consent_given = document.getElementById('consent-given');
+  consent_refused.addEventListener("click", (obj) => {
+    window.close()
```
Comment (ErikBjare) [inline static/consent.js:7]: I think this should prompt the user to uninstall the extension, if we can, as per Mozilla guidelines: https://extensionworkshop.com/documentation/develop/best-practices-for-collecting-user-data-consents/

image

File: static/popup.js [hunk 2]
```diff
@@ -3,7 +3,21 @@
...
+    chrome.storage.local.get("noConsentGiven", (obj) => {
+      console.log('noConsentGiven: ', obj.noConsentGiven)
+      if (obj.noConsentGiven) {
+        enabledCheckbox.setAttribute('disabled', '');
+        showConsentBtn.style.display = 'inline-block';
+      } else {
+        enabledCheckbox.removeAttribute('disabled');
+        showConsentBtn.style.display = 'none';
+      }
+    })
```
Comment (ErikBjare) [inline static/popup.js:20]:
```suggestion
    });
```

**Comment by ErikBjare** [comment]
Very nice! Thank you for this @Morpheus0x :heart: 

Can we prompt the user to uninstall the extension if they do not consent?

**Comment by Morpheus0x** [comment]
> Very nice! Thank you for this @Morpheus0x heart
> 
> Can we prompt the user to uninstall the extension if they do not consent?

Happy to help 😃

I don't think that's possible. After quite a bit of googling, I didn't find anything, even though it is referenced twice on the page linked (also at the very bottom image).
The only thing related I found was [this stackoverflow question](https://stackoverflow.com/questions/24669751/uninstall-remove-firefox-extension-programmatically), but the AddonManager API seems deprecated, since no documentation exists. I didn't even try to use this.

That is why I implemented an additional safeguard if the user closes the popup and then tries to enable the watcher.

**Comment by Morpheus0x** [comment]
Got a very quick response from discourse and implemented remove extension. My safeguard is still valid, since the popup could just be closed, which doesn't remove the extension.

**Comment by Morpheus0x** [comment]
This should be my last commit for this feature. I forgot to add support to skip the consent window by setting the enterprise policy, see [Adding policy support to your extension](https://extensionworkshop.com/documentation/enterprise/adding-policy-support-to-your-extension/) for info. I have added the needed enterprise policy to the readme.

**Comment by Morpheus0x** [comment]
Any timeline for merging this PR? I would like to start working on #82 and don't want to get into any merge conflicts

**Comment by ErikBjare** [comment]
I'm on vacation, but can merge it straight away in the meantime so you can keep working :)

**Comment by ErikBjare** [comment]
Thanks again for working on this @Morpheus0x ❤️

**Comment by Morpheus0x** [comment]
> I'm on vacation, but can merge it straight away in the meantime so you can keep working :)

Awesome, thanks, have a relaxing time.

> Thanks again for working on this @Morpheus0x heart

Sure, no problem :wink:

**Comment by GeorchW** [comment]
Is the extension going to re-appear in the extension store soon then? Can I do anything to help with this?

**Pull request merged**